components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -612,8 +612,22 @@ public static void sendToRetryPage(HttpServletRequest request, HttpServletRespon
             URIBuilder uriBuilder = new URIBuilder(
                     ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL());
             if (status != null && statusMsg != null) {
-                uriBuilder.addParameter("status", status);
-                uriBuilder.addParameter("statusMsg", statusMsg);
+                if (context != null) {
+                    Map<String, String> failureData = new HashMap<>();
+                    failureData.put("status", status);
```

Comment:
Add constants

Reply:
Updated existing `STATUS` and `STATUS_MSG` values in `FrameworkConstants` and used those here.